src/IO/CFS/WriteBufferFromCFS.h

```diff
@@ -0,0 +1,50 @@
+#pragma once
+
+#include "config.h"
+#include <IO/WriteBuffer.h>
+#include <IO/WriteSettings.h>
+#include <IO/WriteBufferFromFileBase.h>
+#include <Poco/Util/AbstractConfiguration.h>
+#include <fcntl.h>
+#include <string>
+#include <memory>
+
+
+#ifndef O_DIRECT
+#define O_DIRECT 00040000
+#endif
+
+namespace DB
+{
+/**
+ * Accepts CFS path to file and opens it.
+ * Closes file by himself (thus "owns" a file descriptor).
+ */
+class WriteBufferFromCFS final :  public WriteBufferFromFileBase
```

Comment:
What are the differences with `WriteBufferFromFileDescriptor`? Can you please describe them?

Also I've noticed that this implementation does not have profile events, like `WriteBufferFromFileDescriptor` (i.e. `ProfileEvents::WriteBufferFromFileDescriptorWriteFailed`)